internal/webhooks/clusterclass.go

```diff
@@ -154,20 +163,226 @@ func (webhook *ClusterClass) validate(ctx context.Context, old, new *clusterv1.C
 		}
 
 		// Ensure no MachineDeploymentClass currently in use has been removed from the ClusterClass.
-		allErrs = append(allErrs, webhook.validateRemovedMachineDeploymentClassesAreNotUsed(clusters, old, new)...)
-	}
-	// Validate variables.
-	allErrs = append(allErrs, variables.ValidateClusterClassVariables(new.Spec.Variables, field.NewPath("spec", "variables"))...)
+		allErrs = append(allErrs,
+			webhook.validateRemovedMachineDeploymentClassesAreNotUsed(clusters, old, new)...)
 
-	// Validate patches.
-	allErrs = append(allErrs, validatePatches(new)...)
+		// Ensure no Variable would be invalidated by the update in spec
+		allErrs = append(allErrs,
+			validateVariableUpdates(clusters, old, new, field.NewPath("spec", "variables"))...)
+	}
 
 	if len(allErrs) > 0 {
 		return apierrors.NewInvalid(clusterv1.GroupVersion.WithKind("ClusterClass").GroupKind(), new.Name, allErrs)
 	}
 	return nil
 }
 
+// validateVariableUpdates checks if the updates made to ClusterClassVariables are valid.
+// It retrieves a list of variables of interest and validates:
+// 1) Altered Variables defined on any exiting Cluster are still valid with the updated Schema.
```

Comment:
The gap between godoc and implementation is that the variables weren't altered (or added in 4.), their corresponding ClusterClassVariable / schema was altered / added.

But I don't want to nitpick to much here, but I think as this func is fairly complicated it would be good that readers get a clear picture. Up to you :)

Reply:
Elegant solution :). I somehow didn't think of that one :D